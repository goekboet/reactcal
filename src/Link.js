import React from 'react'

const Link = props => {
    const handleClick = f => {
        return e => {
            e.preventDefault();
            f();
        }
    }
    return (<a href="#" onClick={handleClick(props.action)} >{props.text}</a>);
}

Link.propTypes = {
    text: React.PropTypes.string.isRequired,
    action: React.PropTypes.func
}

export default Link;