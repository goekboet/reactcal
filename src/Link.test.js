import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Link from './Link';

describe('Link view', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Link text="test" action={x => x} />, div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<Link text="test" action={x => x} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('calls action', () => {
        const preventDefault = jest.fn();
        const action = jest.fn();
        const link = shallow(<Link text="test" action={action} />);

        link.find('a').simulate('click', { preventDefault });
        expect(preventDefault).toBeCalled();
        expect(action).toBeCalled();
    })
});