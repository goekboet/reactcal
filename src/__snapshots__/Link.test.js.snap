exports[`Link view matches snapshot 1`] = `
<a
  href="#"
  onClick={[Function]}>
  test
</a>
`;
